functions/activate-batch-background.js

```diff
@@ -50,7 +66,7 @@ async function handle(data, db, contract) {
     for(const account of existingAddresses) {
       const count = Math.min(account.balance ?? 0, postRH[account.address] ?? 0)
       holderBalance[account.address] = count
-      for(let i = 0;i < count;i++) {
+      for(let i = 0; i < count + addressBadluckCount[account.address].badLuckCount; i++) {
```

Comment:
Need to account for non-existent account (first time user), here and everywhere else that accesses `addressBadluckCount`. You can do something up top maybe and intialise it first with all the addresses in the batch, then fill it in later.